src/template.py

```diff
@@ -12,11 +12,38 @@ def render_template(logger, template):
     jinja_template = Environment(loader=BaseLoader()).from_string(template)
     jinja_template.globals.update({
         "bitwarden_lookup": BitwardenLookupHandler(logger).bitwarden_lookup,
+        "bitwarden_lookup_with_name": BitwardenLookupHandler(logger).bitwarden_lookup_with_name,
     })
     return jinja_template.render()
 
 
-def create_template_secret(logger, secret, filename, template):
+def create_template_secret(logger, spec, body, **kwargs):
+    template = spec.get('template')
+    filename = spec.get('filename')
+    secret_name = spec.get('name')
+    labels = spec.get('labels')
+
+    annotations = {
+        "managed": "bitwarden-template.lerentis.uploadfilter24.eu",
+        "managedObject": f"{body.get('metadata').get('namespace')}/{body.get('metadata').get('name')}"
+    }
+
+    if not labels:
+        labels = {}
+
+    owner_references = [{
+        "apiVersion": f"{body.get('apiVersion')}",
+        "blockOwnerDeletion": True,
```

Comment:
what is the intention behind `blockOwnerDeletion` and `controller`?

Reply:
I don't want the [owner to be deleted unless this is also deleted](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/object-meta/#System:~:text=time%20package%20offers.-,ownerReferences,-(%5B%5DOwnerReference)).

And I also want the generated secret to be deleted if the bitwarden-templates.lerentis object is deleted.